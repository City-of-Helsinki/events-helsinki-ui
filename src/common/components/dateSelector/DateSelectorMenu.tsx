import classNames from 'classnames';
import { IconAngleLeft, IconAngleRight, IconCalendarPlus } from 'hds-react';
import React, { ChangeEvent, FunctionComponent, MutableRefObject } from 'react';
import { useTranslation } from 'react-i18next';

import { translateValue } from '../../../util/translateUtils';
import Checkbox from '../checkbox/Checkbox';
import DateRangePicker from '../dateRangePicker/DateRangePicker';
import styles from './dateSelectorMenu.module.scss'; // the locale you want

export const testIds = {
  menu: 'date-selector-menu',
};

interface Props {
  backBtnRef?: MutableRefObject<HTMLButtonElement | null>;
  customDatesBtnRef?: MutableRefObject<HTMLButtonElement | null>;
  dateTypes: string[];
  dateTypeOptions: string[];
  endDate: Date | null;
  isCustomDate: boolean;
  isOpen: boolean;
  name: string;
  onChangeDateTypes: (value: string[]) => void;
  onChangeEndDate: (date: Date | null) => void;
  onChangeStartDate: (date: Date | null) => void;
  onCloseMenu: () => void;
  startDate: Date | null;
  toggleIsCustomDate: () => void;
}

const DateSelectorMenu: FunctionComponent<Props> = ({
  backBtnRef,
  customDatesBtnRef,
  dateTypes,
  dateTypeOptions,
  endDate,
  isCustomDate,
  isOpen,
  name,
  onChangeDateTypes,
  onChangeEndDate,
  onChangeStartDate,
  onCloseMenu,
  startDate,
  toggleIsCustomDate,
}) => {
  const { t } = useTranslation();

  const handleCheckboxChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (dateTypes.indexOf(event.target.value) !== -1) {
      onChangeDateTypes(
        dateTypes.filter((item) => item !== event.target.value)
      );
    } else {
      onChangeDateTypes([...dateTypes, event.target.value]);
    }
  };

  if (!isOpen) return null;

  return (
    <div
      data-testid={testIds.menu}
      className={classNames(styles.dateSelectorMenu, {
        [styles.isCustomDate]: isCustomDate,
      })}
    >
      {!isCustomDate && (
        <div className={styles.checkboxWrapper}>
          {dateTypeOptions.map((option) => {
            return (
              <Checkbox
                key={option}
                checked={dateTypes.indexOf(option) !== -1}
                id={`name_${option}`}
                label={translateValue(
                  'commons.dateSelector.dateType',
                  option,
                  t
                )}
                name={name}
                onChange={handleCheckboxChange}
                value={option}
              />
            );
          })}
        </div>
      )}

      <button
        ref={customDatesBtnRef}
        className={classNames(styles.button, styles.btnSelectDates, {
          [styles.hidden]: isCustomDate,
        })}
        onClick={toggleIsCustomDate}
        type="button"
      >
        <IconCalendarPlus aria-hidden />
        <div className={styles.buttonText}>
          {t('commons.dateSelector.menu.buttonCustom')}
        </div>
        <IconAngleRight aria-hidden />
      </button>

      <button
        ref={backBtnRef}
        className={classNames(styles.button, styles.btnBack, {
          [styles.hidden]: !isCustomDate,
        })}
        onClick={toggleIsCustomDate}
        type="button"
      >
        <IconAngleLeft aria-hidden />
        <div className={styles.buttonText}>
          {t('commons.dateSelector.menu.buttonBack')}
        </div>
      </button>

      {isCustomDate && (
        <div className={styles.wrapper}>
          <DateRangePicker
            endDate={endDate}
            onChangeEndDate={onChangeEndDate}
            onChangeStartDate={onChangeStartDate}
            startDate={startDate}
          />
        </div>
      )}
      <button
        className={classNames(styles.button, styles.btnClose, {
          [styles.hidden]: !isCustomDate,
        })}
        onClick={onCloseMenu}
        type="button"
      >
        <div className={styles.buttonText}>
          {t('commons.dateSelector.menu.buttonClose')}
        </div>
      </button>
    </div>
  );
};

export default DateSelectorMenu;
