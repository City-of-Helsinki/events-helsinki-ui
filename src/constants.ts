export const CATEGORIES = {
  CULTURE: "culture",
  DANCE: "dance",
  FOOD: "food",
  INFLUENCE: "influence",
  MISC: "misc",
  MOVIE: "movie",
  MUSEUM: "museum",
  MUSIC: "music",
  NATURE: "nature",
  SPORT: "sport",
  THEATRE: "theatre"
};

export const DISTRICTS = {
  ALPPIHARJU: "kaupunginosa:alppiharju",
  EIRA: "kaupunginosa:eira",
  ETU_TÖÖLÖ: "kaupunginosa:etu-t%C3%B6%C3%B6l%C3%B6",
  HAAGA: "kaupunginosa:haaga",
  HERMANNI: "kaupunginosa:hermanni",
  HERTTONIEMI: "kaupunginosa:herttoniemi",
  KAARELA: "kaupunginosa:kaarela",
  KAARTINKAUPUNKI: "kaupunginosa:kaartinkaupunki",
  KAIVOPUISTO: "kaupunginosa:kaivopuisto",
  KALLIO: "kaupunginosa:kallio",
  KAMPPI: "kaupunginosa:kamppi",
  KATAJANOKKA: "kaupunginosa:katajanokka",
  KLUUVI: "kaupunginosa:kluuvi",
  KONALA: "kaupunginosa:konala",
  KOSKELA: "kaupunginosa:koskela",
  KRUUNUNHAKA: "kaupunginosa:kruununhaka",
  KULOSAARI: "kaupunginosa:mustikkamaa-korkeasaari,kaupunginosa:kulosaari",
  KUMPULA: "kaupunginosa:kumpula",
  KÄPYLÄ: "kaupunginosa:k%C3%A4pyl%C3%A4",
  LAAJASALO:
    "kaupunginosa:laajasalo,kaupunginosa:villinki,kaupunginosa:santahamina",
  LAAKSO: "kaupunginosa:laakso",
  LAUTTASAARI: "kaupunginosa:lauttasaari",
  LÄNSISATAMA: "kaupunginosa:l%C3%A4nsisatama",
  MALMI: "kaupunginosa:malmi",
  MEILAHTI: "kaupunginosa:meilahti",
  MELLUNKYLÄ: "kaupunginosa:mellunkyl%C3%A4",
  MUNKKINIEMI: "kaupunginosa:munkkiniemi",
  OULUNKYLÄ: "kaupunginosa:oulunkyl%C3%A4",
  PAKILA: "kaupunginosa:pakila",
  PITÄJÄMÄKI: "kaupunginosa:pit%C3%A4j%C3%A4nm%C3%A4ki",
  PUKINMÄKI: "kaupunginosa:pukinm%C3%A4ki",
  PUNAVUORI: "kaupunginosa:punavuori",
  RUSKEASUO: "kaupunginosa:ruskeasuo",
  SUOMENLINNA: "kaupunginosa:suomenlinna,kaupunginosa:ulkosaaret",
  SUURMETSÄ: "kaupunginosa:suurmets%C3%A4",
  SUUTARILA: "kaupunginosa:suutarila",
  SÖRNÄINEN: "kaupunginosa:s%C3%B6rn%C3%A4inen",
  TAKA_TÖÖLÖ: "kaupunginosa:taka-t%C3%B6%C3%B6l%C3%B6",
  TAMMISALO: "kaupunginosa:tammisalo",
  TAPANINKYLÄ: "kaupunginosa:tapaninkyl%C3%A4",
  ULLANLINNA: "kaupunginosa:ullanlinna",
  VALLILA: "kaupunginosa:vallila",
  VANHAKAUPUNKI: "kaupunginosa:vanhakaupunki",
  VARTIOKYLÄ: "kaupunginosa:vartiokyl%C3%A4",
  VARTIOSAARI: "kaupunginosa:vartiosaari",
  VIIKKI: "kaupunginosa:viikki",
  VUOSAARI: "kaupunginosa:vuosaari",
  ÖSTERSUNDOM:
    "kaupunginosa:%C3%B6stersundom,kaupunginosa:salmenkallio,kaupunginosa:talosaari,kaupunginosa:karhusaari,kaupunginosa:ultuna"
};

export const DATE_TYPES = {
  ALL: "all",
  THIS_WEEK: "this_week",
  TODAY: "today",
  TOMORROW: "tomorrow",
  WEEKEND: "weekend"
};

export const KEYWORD_TYPES = {
  DISTRICT: "district",
  KEYWORD: "keyword",
  PLACE: "place",
  YSO: "yso"
};

export enum SUPPORT_LANGUAGES {
  FI = "fi",
  SV = "sv",
  EN = "en"
}

export const TARGET_GROUPS = {
  CHILDREN: "children",
  YOUNG_PEOPLE: "young_people"
};
