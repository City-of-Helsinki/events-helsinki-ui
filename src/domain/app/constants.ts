export enum ROUTES {
  ABOUT = '/about',
  COLLECTION = '/collection/:id',
  COLLECTIONS = '/collections',
  EVENTS = '/events',
  EVENT = '/event/:id',
  HOME = '/home',
  HOME_PREVIEW = '/home/:id',
}
