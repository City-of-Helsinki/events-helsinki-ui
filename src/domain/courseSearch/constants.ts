export const DEFAULT_SEARCH_FILTERS = {
  categories: [],
  dateTypes: [],
  divisions: [],
  end: null,
  isFree: false,
  onlyOngoingCourses: false,
  places: [],
  start: null,
  text: [],
};
