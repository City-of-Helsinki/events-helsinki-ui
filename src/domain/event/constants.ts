export const mockEventData = {
  eventDetails: {
    description: {
      en: "description en",
      fi: "description fi",
      sv: "description sv"
    },
    endTime: null,
    externalLinks: [
      {
        link: "https://www.facebook.com/events/12345687/",
        name: "extlink_facebook"
      }
    ],
    id: "123",
    images: [
      {
        id: "60578",
        name: "piha-2.jpg",
        url: "https://api.hel.fi/linkedevents/media/images/piha-2_yYlCdPB.jpg"
      }
    ],
    inLanguage: [
      {
        id: "fi",
        name: {
          en: "Finnish",
          fi: "suomi",
          sv: "finska"
        },
        translationAvailable: true
      }
    ],
    infoUrl: {
      en:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080",
      fi:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080",
      sv:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080"
    },
    keywords: [],
    location: {
      addressLocality: {
        en: "locality en",
        fi: "locality fi",
        sv: "locality sv"
      },
      divisions: [],
      email: "location@info.com",
      infoUrl: {
        en: "http://www.infourl.com",
        fi: "http://www.infourl.com",
        sv: "http://www.infourl.com"
      },
      name: {
        en: "name en",
        fi: "name fi",
        sv: "name sv"
      },
      position: {
        coordinates: [25.01497, 60.250507],
        type: "Point"
      },
      postalCode: "01234",
      streetAddress: {
        en: "streetAddress en",
        fi: "streetAddress fi",
        sv: "streetAddress sv"
      },
      telephone: {
        en: "044 123 4567",
        fi: "044 123 4567",
        sv: "044 123 4567"
      }
    },
    name: {
      en: "name en",
      fi: "name fi",
      sv: "name sv"
    },
    offers: [
      {
        description: null,
        infoUrl: null,
        isFree: true,
        price: null
      }
    ],
    provider: null,
    publisher: "provider:123",
    shortDescription: {
      en: "short description en",
      fi: "short description fi",
      sv: "short description sv"
    },
    startTime: "2019-12-13T08:00:00Z"
  }
};

// Amount of the similar events
export const SIMILAR_EVENTS_AMOUNT = 8;

export const EVENT_KEYWORD_BLACK_LIST = [
  "helmet:10592", // Yleinen
  "helmet:10689", // Helsinki
  "helmet:11916", // Infonäytöt (Helsinki)
  "kulke:53", // Suomi
  "kulke:54", // Ruotsi
  "kulke:55", // Englanti
  "kulke:616", // Näkyy TV-sovelluksessa
  "kulke:671", // Normaali
  "kulke:673" // Suomi (TV)
];
