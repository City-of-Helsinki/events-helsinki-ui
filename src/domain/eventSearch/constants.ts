// Page size of the event list
export const PAGE_SIZE = 10;

export enum EVENT_SORT_OPTIONS {
  DURATION = 'duration',
  DURATION_DESC = '-duration',
  END_TIME = 'end_time',
  END_TIME_DESC = '-end_time',
  LAST_MODIFIED_TIME = 'last_modified_time',
  LAST_MODIFIED_TIME_DESC = '-last_modified_time',
  START_TIME = 'start_time',
  START_TIME_DESC = '-start_time',
}

export enum EVENT_SEARCH_FILTERS {
  CATEGORIES = 'categories',
  DATE_TYPES = 'dateTypes',
  DIVISIONS = 'divisions',
  END = 'end',
  IS_FREE = 'isFree',
  KEYWORD_NOT = 'keywordNot',
  KEYWORDS = 'keywords',
  ONLY_CHILDREN_EVENTS = 'onlyChildrenEvents',
  PAGE = 'page',
  PLACES = 'places',
  PUBLISHER = 'publisher',
  START = 'start',
  TEXT = 'text',
}

export const CULTURE_KEYWORDS = [
  'kulke:33', // Teatteri
  'kulke:51', // Sirkus
  'kulke:205', // Elokuva ja media
  'kulke:351', // Teatteri ja sirkus
  'matko:teatteri', // teatteri
  'yso:p360', // cultural events
  'yso:p1235', // films
  'yso:p1278', // dance (performing arts)
  'yso:p1808', // music
  'yso:p2625', // in Finnish teatteritaide, "theatre arts"
  'yso:p2739', // fine arts
  'yso:p2850', // performing arts
  'yso:p2851', // art
  'yso:p4934', // museums
  'yso:p5121', // exhibitions
  'yso:p6889', // art exhibitions
  'yso:p7969', // literary art
  'yso:p8113', // literature
  'yso:p8144', // art museums
  'yso:p9592', // modern art
  'yso:p9593', // contemporary art
  'yso:p10105', // contemporary dance
  'yso:p16327', // cinema (art forms)
];

export const INFLUENCE_KEYWORDS = [
  'yso:p1657', // Vaikuttaminen
  'yso:p10727', // Osallistuminen
];

export const MUSEUM_KEYWORDS = [
  'matko:museo', // Museo
  'yso:p4934', // Museot
];
