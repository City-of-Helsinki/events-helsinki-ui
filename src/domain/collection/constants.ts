import { mockEventData } from "../event/constants";

export const mockCollection = {
  collectionDetails: {
    curatedEvents: [
      `http://localhost:3000/fi/event/${mockEventData.eventDetails.id}`
    ],
    curatedEventsTitle: {
      en: "At least visit these",
      fi: "Käy ainakin näissä",
      sv: "Besök åtminstone dessa"
    },
    description: {
      en:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      fi:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      sv:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus."
    },
    eventListQuery:
      "http://localhost:3000/en/events?categories=music&districts=kaupunginosa%3Aetu-t%C3%B6%C3%B6l%C3%B6",
    eventListTitle: {
      en: "All the best events of the fall",
      fi: "Kaikki syksyn parhaat tapahtumat",
      sv: "Höstens bästa händelser"
    },
    id: "1",
    linkText: {
      en: "Read more on the project website",
      fi: "Lue lisää hankkeen omilta sivuilta",
      sv: "Läs mer på projektets webbplats"
    },
    linkUrl: {
      en: "http://www.google.com",
      fi: "http://www.google.com",
      sv: "http://www.google.com"
    },
    similarCollectionsTitle: {
      en: "Check also these",
      fi: "Katso myös nämä",
      sv: "Se dessa också"
    },
    socialMediaDescription: {
      en: "We put together the best foliage for the fall",
      fi: "Kokosimme parhaat tärpit syksylle",
      sv: "Vi sätter ihop det bästa bladverket för hösten"
    },
    subtitles: {
      en: "Subtitles en",
      fi: "Subtitles fi",
      sv: "Subtitles sv"
    },
    title: {
      en: "The most bustling tricks of the fall",
      fi: "Syksyn riehakkaimmat riennot",
      sv: "Höstens mest livliga knep"
    }
  }
};

// Amount of the similar collections
export const SIMILAR_COLLECTIONS_AMOUNT = 8;
