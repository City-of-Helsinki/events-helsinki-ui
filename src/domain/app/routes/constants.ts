import { MAPPED_PLACES } from '../../eventSearch/constants';

// Remember also update static urls to updateSitemap
export type ROUTE_TYPES = {
  ABOUT: '/about';
  ACCESSIBILITY: '/accessibility';
  COLLECTION: '/collection/:slug';
  COLLECTIONS: '/collections';
  EVENTS: '/events';
  EVENT: '/event/:id';
  COURSES: '/courses';
  COURSE: '/course/:id';
  HOME: '/home';
  HOME_PREVIEW: '/home/:id';
  EVENT_PLACE: string;
};

export const ROUTES: ROUTE_TYPES = {
  ABOUT: '/about',
  ACCESSIBILITY: '/accessibility',
  COLLECTION: '/collection/:slug',
  COLLECTIONS: '/collections',
  EVENTS: '/events',
  EVENT: '/event/:id',
  COURSES: '/courses',
  COURSE: '/course/:id',
  HOME: '/home',
  HOME_PREVIEW: '/home/:id',
  EVENT_PLACE: `/:place(${Object.keys(MAPPED_PLACES).join('|')})`,
};
