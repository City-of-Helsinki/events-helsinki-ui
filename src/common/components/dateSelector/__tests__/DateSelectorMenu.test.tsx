import * as React from 'react';

import { DATE_TYPES } from '../../../../constants';
import { render } from '../../../../util/testUtils';
import DateSelectorMenu from '../DateSelectorMenu';

test('matches snapshot', () => {
  const { container } = render(
    <DateSelectorMenu
      dateTypes={['type1', 'type2']}
      dateTypeOptions={[
        DATE_TYPES.TODAY,
        DATE_TYPES.TOMORROW,
        DATE_TYPES.THIS_WEEK,
        DATE_TYPES.WEEKEND,
      ]}
      endDate={new Date('2019-09-31')}
      isCustomDate={false}
      isOpen={true}
      name="date"
      onChangeDateTypes={jest.fn()}
      onChangeEndDate={jest.fn()}
      onChangeStartDate={jest.fn()}
      onCloseMenu={jest.fn()}
      startDate={new Date('2019-08-01')}
      toggleIsCustomDate={jest.fn()}
    />
  );

  expect(container.firstChild).toMatchSnapshot();
});
