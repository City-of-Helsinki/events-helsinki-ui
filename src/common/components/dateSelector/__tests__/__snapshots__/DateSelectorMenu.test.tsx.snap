// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="dateSelectorMenu"
  data-testid="date-selector-menu"
>
  <div
    class="checkboxWrapper"
  >
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
      style="--background-unselected: var(--color-white); line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_today"
        name="date"
        type="checkbox"
        value="today"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_today"
      >
        Tänään
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
      style="--background-unselected: var(--color-white); line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_tomorrow"
        name="date"
        type="checkbox"
        value="tomorrow"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_tomorrow"
      >
        Huomenna
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
      style="--background-unselected: var(--color-white); line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_this_week"
        name="date"
        type="checkbox"
        value="this_week"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_this_week"
      >
        Tällä viikolla
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
      style="--background-unselected: var(--color-white); line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_weekend"
        name="date"
        type="checkbox"
        value="weekend"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_weekend"
      >
        Viikonloppuna
      </label>
    </div>
  </div>
  <button
    class="button btnSelectDates"
    type="button"
  >
    <svg
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h24v24H0z"
        />
        <path
          d="M20 14v3h3v2h-3v3h-2v-3h-3v-2h3v-3h2zM17 2a1 1 0 011 1v1h4v8h-2v-1H4v8h9v2H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 4H4v3h16V6z"
          fill="currentColor"
        />
      </g>
    </svg>
    <div
      class="buttonText"
    >
      Valitse päivät
    </div>
    <svg
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <polygon
          points="0 24 0 0 24 0 24 24"
        />
        <polygon
          fill="currentColor"
          points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
        />
      </g>
    </svg>
  </button>
  <button
    class="button btnBack hidden"
    type="button"
  >
    <svg
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <polygon
          points="24 0 24 24 0 24 0 0"
        />
        <polygon
          fill="currentColor"
          points="10.5 12 15.5 17 14 18.5 7.5 12 14 5.5 15.5 7"
        />
      </g>
    </svg>
    <div
      class="buttonText"
    >
      Takaisin
    </div>
  </button>
  <button
    class="button btnClose hidden"
    type="button"
  >
    <div
      class="buttonText"
    >
      Sulje
    </div>
  </button>
</div>
`;
